src/community/wps-download/src/test/java/org/geoserver/wps/gs/download/DownloadProcessTest.java

```diff
@@ -1674,6 +1690,185 @@ public void testWrongOutputFormat() throws Exception {
+     * 9km_32610.tif with resolution = 8712.564801039759900 meters
+     * 
+     */
+    @Test
+    public void testDownloadGranuleNativeResolution() throws Exception {
+
+        DownloadEstimatorProcess limits = new DownloadEstimatorProcess(
+                new StaticDownloadServiceConfiguration(), getGeoServer());
+        final WPSResourceManager resourceManager = getResourceManager();
+        DownloadProcess downloadProcess = new DownloadProcess(getGeoServer(), limits,
+                resourceManager);
+
+        
+        // Setting filter to get the granule with resolution
+        final PropertyName property = FeatureUtilities.DEFAULT_FILTER_FACTORY
+                .property("resolution");
+        Filter filter = (Filter) FeatureUtilities.DEFAULT_FILTER_FACTORY.greaterOrEqual(property,
+                FeatureUtilities.DEFAULT_FILTER_FACTORY.literal(16000));
+        
+        File rasterZip = downloadProcess.execute(getLayerId(MIXED_RES), // layerName
+                filter, // filter
+                "image/tiff", // outputFormat
+                null, // targetCRS
+                CRS.decode("EPSG:4326", true), // roiCRS
+                ROI2, // roi
+                false, // cropToGeometry
+                null, // interpolation
+                null, // targetSizeX
+                null, // targetSizeY
+                null, // bandSelectIndices
+                null, // Writing params
+                new NullProgressListener() // progressListener
+        );
+
+        // Final checks on the result
+        Assert.assertNotNull(rasterZip);
+        GeoTiffReader reader = null;
+        GridCoverage2D gc = null;
+        try {
+            final File[] tiffFiles = extractTIFFFile(rasterZip);
+            Assert.assertNotNull(tiffFiles);
+            Assert.assertTrue(tiffFiles.length > 0);
+            reader = new GeoTiffReader(tiffFiles[0]);
+            gc = reader.read(null);
+
+            Assert.assertNotNull(gc);
+
+            // check coverage size
+            MathTransform mt = reader.getOriginalGridToWorld(PixelInCell.CELL_CENTER);
+            AffineTransform2D transform2D = (AffineTransform2D) mt;
+            double resX = transform2D.getScaleX();
+            double resY = transform2D.getScaleY();
+
+            // Check the read resolutions is the native one of granule
+            Assert.assertEquals(17550.94845318, resX, 1E-6);
+            Assert.assertEquals(-17550.94845318, resY, 1E-6);
+
+        } finally {
+            if (gc != null) {
+                CoverageCleanerCallback.disposeCoverage(gc);
+            }
+            if (reader != null) {
+                reader.dispose();
+            }
+
+            // clean up process
+            resourceManager.finished(resourceManager.getExecutionId(true));
+        }
+
+        // Download native resolution 2
+        filter = FeatureUtilities.DEFAULT_FILTER_FACTORY.and(FeatureUtilities.DEFAULT_FILTER_FACTORY.lessOrEqual(property,
```

Comment:
This is hard to read, could you put the FeatureUtilities.DEFAULT_FILTER_FACTORY into a local variable? (e..g, ``FilterFactory2 ff = FeatureUtilities.DEFAULT_FILTER_FACTORY``)